File: log2data.py
# -*- coding: utf-8 -*-
#!/usr/bin/env python
import argparse
import sys
import os
import re

def print_sysinfo(logfile, argvstr, cc='#'):
  print(argvstr)
  date = os.popen('date').read().rstrip()
  pwd  = os.popen('pwd').read().rstrip()
  hostname = os.popen('hostname').read().rstrip()
  user = os.popen('whoami').read().rstrip()
  
  print("{0} Generated {1} by {2}\n{0} {3}:{4}\n{0} {5}\n\n".format(cc,date,user,hostname,pwd,argvstr), file=logfile)
  
  debug = False
  if debug:
    print("{0} Generated {1} by {2}\n{0} {3}:{4}\n{0} {5}\n\n".format(cc,date,user,hostname,pwd,argvstr))



def main(argv):
  argvstr = " ".join(argv)
  
  # construct the argument parse and parse the arguments
  ap = argparse.ArgumentParser()
  ap.add_argument("-n", "--network", default="alexnet",
        help="name of network")
  ap.add_argument("-d", "--dataset", default="imagenet",
        help="name of dataset")
  ap.add_argument("-l", "--logfile", default="test.log",
        help="name of logfile")
  ap.add_argument("-o", "--outfile", default="out.log",
        help="name of logfile")
  ap.add_argument("-e", "--epoch", type=int, default=10,
        help="max epoch")
  args = vars(ap.parse_args())

  outfile = open(args["outfile"], 'w', encoding='UTF-8')
  print("args: ",args)
  print_sysinfo(outfile, argvstr, '//')

  START = 'rate   iter_k   epoch  num_m | valid_loss/acc | train_loss/acc | batch_loss/acc |  time'
  END = '========='

  rows = open(args["logfile"], 'r').read()
  result = re.findall('%s(.*)%s' % (START,END), rows)
  print("result: ", result)
  

if __name__ =="__main__":
  debug = True
  if debug:
    print(sys.argv)
    print(sys.argv[1:]) # ignore script name

  main(sys.argv)

File: log2data.rb
# encoding: utf-8
#!/usr/bin/env ruby

require 'optparse'

def print_sysinfo(outfile, argvstr, cc='#')
  print(argvstr)
  date = `date`.chomp
  pwd  = `pwd`.chomp
  hostname = `hostname`.chomp
  user = `whoami`.chomp
 
  outfile.print("#{cc} Generated #{date} by #{user}\n#{cc} #{hostname}:#{pwd}\n#{cc} #{$0} #{argvstr}\n\n")
end

argvstr = ARGV.join(" ")

# This hash will hold all of the options 
# parsed from the command-line by 
# OptionParser. 
options = {}

optparse = OptionParser.new do |opts|
  # Set a banner, displayed at the top 
  # of the help screen. 
  opts.banner = "Usage: #{$0} [options] <in.cdl> <out.cdl>"
  # Define the options, and what they do 
  options[:force] = false
  opts.on( '-f', '--force', 'Force to overwrite output file' ) do
    options[:force] = true
  end
  options[:capval] = 0.0
  opts.on( '-c', '--cap value', Float, 'CPP -> MOMCAPS' ) do|f|
    options[:capval] = f
  end
  options[:bracket] = false
  opts.on( '-b', '--bracket', 'Bracket <> -> []' ) do
    options[:bracket] = true
  end
  options[:dummy] = false
  opts.on( '-d', '--dummy', 'Dummy cap and resistor removal' ) do
    options[:dummy] = true
  end
  options[:logfile] = nil
  opts.on( '-l', '--logfile FILE', 'Write log to FILE' ) do|file|
    options[:logfile] = file
  end
  # This displays the help screen, all programs are 
  # assumed to have this option. 
  opts.on( '-h', '--help', 'Display this screen' ) do
    puts opts
    exit
  end
end

# Parse the command-line. The 'parse' method simply parses 
# ARGV, while the 'parse!' method parses ARGV and removes 
# any options found there, as well as any parameters for 
# the options. What's left is the list of files to resize.


begin optparse.parse!(ARGV)
 rescue OptionParser::InvalidOption => e
 puts e
 puts optparse
 exit
end

puts "Force to overwrite output file" if options[:force]
puts "CPP -> MOMCAPS #{options[:capval]}" if options[:capval]
puts "Bracket <> -> []" if options[:bracket]
puts "Dummy cap & resistor removal" if options[:dummy]
puts "Logging to file #{options[:logfile]}" if options[:logfile]


outfile = File.open(options[:logfile], 'w')
print_sysinfo(outfile, argvstr, '//')
